src/components/interactive-builder/modals/add-form-reference/add-form-reference.modal.tsx

```diff
@@ -0,0 +1,241 @@
+  InlineNotification,
+  Checkbox,
+  CheckboxGroup,
+  FormGroup,
+  RadioButtonGroup,
+  RadioButton,
+  Stack,
+} from '@carbon/react';
+import { type Form as FormType, type Schema, type Page, type Section, type Question } from '@types';
+import { useForms } from '@hooks/useForms';
+import { useClobdata } from '@hooks/useClobdata';
+import styles from './add-form-reference.scss';
+import { showSnackbar } from '@openmrs/esm-framework';
+
+interface AddFormReferenceModalProps {
+  closeModal: () => void;
+  pageIndex: number;
+  schema: Schema;
+  onSchemaChange: (schema: Schema) => void;
+}
+
+const AddFormReferenceModal: React.FC<AddFormReferenceModalProps> = ({
+  closeModal,
+  pageIndex,
+  schema,
+  onSchemaChange,
+}) => {
+  const { t } = useTranslation();
+  const [selectedForm, setSelectedForm] = useState<FormType>(null);
+  const [pages, setPages] = useState<Page[]>([]);
+  const [selectedPage, setSelectedPage] = useState<Page>(null);
+  const [selectedSection, setSelectedSection] = useState<Section>(null);
+  const [excludedQuestions, setExcludedQuestions] = useState<string[]>([]);
+  const { forms, error, isLoading } = useForms();
+  const { clobdata, isLoadingClobdata, clobdataError } = useClobdata(selectedForm);
+
+  const selectForm = useCallback(({ selectedItem }: { selectedItem: FormType }) => {
+    setSelectedForm(selectedItem);
+    setSelectedPage(null);
+    setSelectedSection(null);
+    setExcludedQuestions([]);
+  }, []);
+
+  useEffect(() => {
+    if (clobdata) {
+      setPages(clobdata.pages);
+    }
+  }, [clobdata]);
+
+  const handleSelectedPage = useCallback(({ selectedItem }: { selectedItem: Page }) => {
+    setSelectedPage(selectedItem);
+    setSelectedSection(null);
+    setExcludedQuestions([]);
+  }, []);
+
+  const handleSelectedSections = useCallback((section: Section) => {
+    setSelectedSection(section);
+    setExcludedQuestions([]);
+  }, []);
+
+  const handleExcludedQuestions = useCallback((question: Question, checked: Boolean) => {
+    setExcludedQuestions((prev) => {
+      if (!checked) {
+        return prev.some((q) => q === question.id) ? prev : [...prev, question.id];
+      } else {
+        return prev.filter((q) => q !== question.id);
+      }
+    });
+  }, []);
+
+  const handleUpdateSchema = () => {
+    updateSchema();
+    closeModal();
+  };
+
+  const updateSchema = () => {
+    try {
+      if (!schema.referencedForms?.some((form) => form.formName === selectedForm.name)) {
+        schema.referencedForms
+          ? schema.referencedForms.push({
+              formName: selectedForm.name,
+              alias: selectedForm.name,
+            })
+          : (schema.referencedForms = [
+              {
+                formName: selectedForm.name,
+                alias: selectedForm.name,
+              },
+            ]);
+      }
+      schema.pages[pageIndex].sections.push({
+        label: `${selectedSection.label} (This section is added from ${selectedForm.name})`,
```

Comment:
Its not necessary to add this text to the label, especially since we don't let them edit this. If its a concern that then the user would not know that its a referenced component from the overview in the interactive builder until they expand the accordion, we can add a tag next to the label saying `Referenced Component`.
```suggestion
        label: selectedSection.label,
```